connection_maker.go

```diff
@@ -100,7 +101,9 @@ func (cm *ConnectionMaker) InitiateConnections(peers []string, replace bool) []e
 			host = peer
 			port = "0" // we use that as an indication that "no port was supplied"
 		}
-		if addr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("%s:%s", host, port)); err != nil {
+		if !isAlnum(port) {
+			errors = append(errors, fmt.Errorf("invalid peer name '%s', should just be host[:port]", peer))
+		} else if addr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("%s:%s", host, port)); err != nil {
```

Comment:
That would be beyond the scope of this change.  I don't know why it isn't just `peer`.